### ISSUE TITLE
Enhancement: Add "Preferred PNI IPv6 MTU" to "net" records

### ISSUE THREAD

**Issue by blakedot** [peeringdb/peeringdb#1471]
Managing MTU values on Private Network Interconnect links is a problem space well-suited for PeeringDB.

The suggestion is to add a "Preferred PNI IPv6 MTU" field alongside maxprefix etc. to serve as a configuration reference.

As network equipment vendors tend to have their own specific interpretations of what "interface MTU" means (& not all PNIs are Ethernet-based), the field should specifically reference IPv6 MTU in order to make people think a bit & check their networks before filling in this field (& not just copypasta whatever is configured on their physical interfaces).

When two networks have conflicting preferred MTU values, it is a recommended best-practice that the lower of the two values be configured in order to avoid compatibility problems & MTU issues.  Further negotiations between peers are out of scope for PeeringDB.

**Do you think this feature will require a formal design?**
no, it's just another field on members' "net" records

**Describe alternatives you've considered**
The current status quo is many PNIs either have mismatched MTUs or they're inconsistent over multiple PNIs between a given pair of peer networks.  The current alternative is lots of emails...

**What is the proposed priority?**
not urgent

**Comment by BarbarossaTM** [comment]
I like the general idea of allowing to specify "the MTU". I agree that "the MTU" is a complicated beast wrt "L2 or L3?", "w/ Ethernet headers or w/o?", etc. However specifying an IPv6 MTU seems confusing to me and will likely lead to quite some errors/misconfigurations.

How 'bout calling it "IP MTU", to make sure to convey that the intention is to specify the Layer3 MTU without Ethernet headers? The size of the IPv4/IPv6 header should not be relevant for this IMHO. :-D

**Comment by blakedot** [comment]
> I like the general idea of allowing to specify "the MTU". I agree that "the MTU" is a complicated beast wrt "L2 or L3?", "w/ Ethernet headers or w/o?", etc. However specifying an IPv6 MTU seems confusing to me and will likely lead to quite some errors/misconfigurations.
> 
> How 'bout calling it "IP MTU", to make sure to convey that the intention is to specify the Layer3 MTU without Ethernet headers? The size of the IPv4/IPv6 header should not be relevant for this IMHO. :-D

IP MTU may indeed be clearer.  In any case. I feel it's important to highlight that this does not include the layer 2 headers.

Just to provide folks an idea of one vendor's (JunOS) interpretation of MTU:

show interfaces media detail ae2 | match mtu 
  Link-level type: Ethernet, MTU: 9192
    Protocol inet, MTU: 9178
    Protocol iso, MTU: 9175
    Protocol inet6, MTU: 9178
    Protocol mpls, MTU: 9158, Maximum labels: 5

**Comment by blakedot** [comment]
PeeringDB seems to use "payload MTU" for IXP LAN records, we should probably remain consistent with that...

**Comment by netravnen** [comment]
Payload for me exclude Ethernet headers.

I agree the wording (payload) is very much "formal" and not widely used outside standards (or similar language, to my knowledge).

**Comment by netravnen** [comment]
IP MTU has a much higher chance of being intuitively understood by most users. IMO.

**Comment by blakedot** [comment]
It occurred to me that "IP protocol MTU preferred on PNIs" might be the clearest way to state this.
But anyway we're bikeshedding at this point, the main thing is to get the feature in there :-)

**Comment by adofou** [comment]
One possibility would be to add a "?" next to the field name to give a more detailed explanation?
As it already exists on "Notes" or "IPv4/IPv6 Prefixes". It won't solve the problem, but it may reduce errors.

But the field name should be both intuitive (so that you know exactly what value to set) and not too restrictive (so that you don't think it's "only" for IPv6"). It's true that "Payload MTU" can be confusing, even if it's already in use for IXP.

In Cisco documentation, they talks about "Interface MTU" and "IP MTU". In Juniper documentation, they talks about "Media MTU" and "IP Protocol MTU". In Nokia documentation, they talks about "Network port MTU" and "IP MTU".
So I guess the term "IP MTU" is common enough to be understandable.

**Comment by leovegoda** [comment]
Closing as there's been no comment for over a year

**Comment by grizz** [comment]
At risk of opening the door to more bikeshedding, I don't think we need PNI in there, so can just call it `preferred_ip_mtu` right?

**Comment by martinhannigan** [comment]
I agree. MTU preference isn't PNI specific. preferred_ip_mtu feels right to
me.

**Comment by jwbensley** [comment]
+1 we'd like to see this feature and `preferred_ip_mtu` makes sense to me because like most networks we don't operate separate MTUs per address family.
